tests/integration/components/o-s-s/attributes-panel-test.ts

```diff
@@ -0,0 +1,175 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click, setupOnerror } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import sinon from 'sinon';
+
+module('Integration | Component | o-s-s/attributes-panel', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.icon = 'fa-laptop-code';
+    this.title = 'Title';
+    this.onSave = sinon.stub();
+    this.onCancel = sinon.stub();
+    this.onEdit = sinon.stub();
+  });
+
+  test('it renders', async function (assert) {
+    await render(hbs`<OSS::AttributesPanel @icon={{this.icon}} @title={{this.title}} @onSave={{this.onSave}} />`);
+    assert.dom('.attributes-panel').exists();
+  });
+
+  test('it renders the icon', async function (assert) {
+    await render(hbs`<OSS::AttributesPanel @icon={{this.icon}} @title={{this.title}} @onSave={{this.onSave}} />`);
+    assert.dom('.attributes-panel .fa-laptop-code').exists();
+  });
+
+  test('it renders the title', async function (assert) {
+    await render(hbs`<OSS::AttributesPanel @icon={{this.icon}} @title={{this.title}} @onSave={{this.onSave}} />`);
+    assert.dom('.attributes-panel span').hasText('Title');
+  });
+
+  test('it renders the contextual-action named block', async function (assert) {
+    await render(hbs`
+      <OSS::AttributesPanel @icon={{this.icon}} @title={{this.title}} @onSave={{this.onSave}}>
+        <:contextual-action>
+          <OSS::Button @icon="fa-plus" @square={{true}} />
+        </:contextual-action>
+      </OSS::AttributesPanel>
+    `);
+    assert.dom('.attributes-panel .fa-plus').exists();
+  });
+
+  module('for view mode', (hooks) => {
+    hooks.beforeEach(async function () {
+      await render(hbs`
+        <OSS::AttributesPanel @icon={{this.icon}} @title={{this.title}} @onSave={{this.onSave}} @onEdit={{this.onEdit}}>
+          <:view-mode>
+            <div class="custom-view-mode">View mode</div>
+          </:view-mode>
+        </OSS::AttributesPanel>
+      `);
+    });
```

Comment:
I think @OwenCoogan started a trend with component rendering in the hooks.

as discussed [in this slack thread](https://upfluence.slack.com/archives/C02FBD3QAQ0/p1698221992007299) ; I think we should keep the methods in the hooks for service / variable initialization and not have renders here.

If we want to spare some lines of code, I think we should better create a private function with a proper naming which makes the render. [Here's an example of that](https://github.com/upfluence/ember-influencer/blob/master/tests/integration/components/discount-code/edition-test.ts#L159)

IMO, we are losing readability in the tests when moving the render inside the hooks ; it might not be a big deal here because we only have 2 test cases but imagine if we had 10+ test cases without any notion of which component is rendered inside each test ?

Reply:
+1.

If I can add something, I would say that given that most of the time we try to have a Given/When/Then way of writing the tests, this means that we are doing Given and When together :/ So yeaa, keeping hooks to "setting up the context" seems better to me. and the extra private method is not much to write AND can even take extra args for even more setup if needed.

Reply:
Ok ok no problem, I will update that